modules/solid_mechanics/src/materials/CompositePowerLawCreepStressUpdate.C

```diff
@@ -0,0 +1,189 @@
+  // set switching functions material properties for each phase
+  for (unsigned int i = 0; i < _num_materials; ++i)
+  {
+    _switchingFunc[i] =
+        &this->template getGenericMaterialProperty<Real, is_ad>(_switchingFuncNames[i]);
+  }
+}
+
+template <bool is_ad>
+void
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeStressInitialize(
+    const GenericReal<is_ad> & effective_trial_stress,
+    const GenericRankFourTensor<is_ad> & elasticity_tensor)
+{
+  RadialReturnStressUpdateTempl<is_ad>::computeStressInitialize(effective_trial_stress,
+                                                                elasticity_tensor);
+  _exp_time = std::pow(_t - _start_time, _m_exponent);
+}
+
+template <bool is_ad>
+template <typename ScalarType>
+ScalarType
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeResidualInternal(
+    const GenericReal<is_ad> & effective_trial_stress, const ScalarType & scalar)
+{
+  const ScalarType stress_delta = effective_trial_stress - _three_shear_modulus * scalar;
+  ScalarType creep_rate =
+      _coefficient[0] * std::pow(stress_delta, _n_exponent[0]) * (*_switchingFunc[0])[_qp] *
+      std::exp(-_activation_energy[0] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+  for (unsigned int n = 1; n < _num_materials; ++n)
+  {
+    creep_rate +=
+        _coefficient[n] * std::pow(stress_delta, _n_exponent[n]) * (*_switchingFunc[n])[_qp] *
+        std::exp(-_activation_energy[n] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+  }
+  return creep_rate * _dt - scalar;
+}
+
+template <bool is_ad>
+GenericReal<is_ad>
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeDerivative(
+    const GenericReal<is_ad> & effective_trial_stress, const GenericReal<is_ad> & scalar)
+{
+  const GenericReal<is_ad> stress_delta = effective_trial_stress - _three_shear_modulus * scalar;
+  GenericReal<is_ad> creep_rate_derivative =
+      -_coefficient[0] * _three_shear_modulus * _n_exponent[0] *
+      std::pow(stress_delta, _n_exponent[0] - 1.0) * (*_switchingFunc[0])[_qp] *
+      std::exp(-_activation_energy[0] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+
+  for (unsigned int n = 1; n < _num_materials; ++n)
+  {
+    creep_rate_derivative +=
+        -_coefficient[n] * _three_shear_modulus * _n_exponent[n] *
+        std::pow(stress_delta, _n_exponent[n] - 1.0) * (*_switchingFunc[n])[_qp] *
+        std::exp(-_activation_energy[n] / (_gas_constant * (*_temperature)[_qp])) * _exp_time;
+  }
+  return creep_rate_derivative * _dt - 1.0;
+}
+
+template <bool is_ad>
+Real
+CompositePowerLawCreepStressUpdateTempl<is_ad>::computeStrainEnergyRateDensity(
+    const GenericMaterialProperty<RankTwoTensor, is_ad> & stress,
+    const GenericMaterialProperty<RankTwoTensor, is_ad> & strain_rate)
+{
+  GenericReal<is_ad> interpolated_exponent = 0.0;
+  for (unsigned int n = 0; n < _num_materials; ++n)
+  {
+    interpolated_exponent += (_n_exponent[n] / (_n_exponent[n] + 1)) * (*_switchingFunc[n])[_qp];
```

Comment:
Probably worth mentioning that teh switching functions must always sum up to 1.0

Reply:
I dont think the switching functions adding up to 1.0 matters in this context because we need to turn off the n_exponent that doesn't belong to a particular phase, and smoothly interpolate it at the interface.

Reply:
It does matter. If someone specifies a (smooth) switching function that goes from 0 to 100 you will see why.

Reply:
One of the ideas is to actually set no plasticity to a particular phase, so one of my working examples (not in test) actually have the interpolation not add up to 1.0 for all elements, essentially acting like weights. In the case of using this as a weighting function similar to MultiplePowerLawCreepStressUpdate material, we can also encounter a problem because the same material can have 2 or 3 different creep behavior, and if set to the same switching functions, then we have the sum of switching functions to be more than 1.0 for those phase. In the case where user do use it a little differently than typical switching function, what do you suggest adding to the material.

Reply:
I have added a regression test that includes two cases in one: Phase 1 has two plasticity laws with the same switching function, and Phase 2 has no plasticity (by not giving it any plasticity parameter). This test should demonstrate that switching functions can be used to selectively pick which phase behave plastically, which phase has multiple plasticity rules, and which phase has no plasticity altogether.